### DISCUSSION TITLE
got error when capturing video from rtsp

### DISCUSSION THREAD

**Discussion by v479038280** [zauberzeug/nicegui#3932] [Q&A]
### Question

I got this error when running the [opencv_webcam](https://github.com/zauberzeug/nicegui/tree/main/examples/opencv_webcam) with streams from rtsp
`video_capture = cv2.VideoCapture("rtsp://admin:password@192.168.1.4:554/Streaming/Channels/101")`

```
NiceGUI ready to go on http://localhost:8080, and http://192.168.1.113:8080
[hevc @ 000001dd9c4f7f00] CABAC_MAX_BIN : 7
[hevc @ 000001dd9c4f7f00] The cu_qp_delta -1094995529 is outside the valid range [-26, 25].
```
Some times there is no error in the terminal,but the picture just stuck there

`NiceGUI ready to go on http://localhost:8080, and http://192.168.1.113:8080
`

![微信截图_20241031015923](https://github.com/user-attachments/assets/d7b5adf0-c81d-4e23-8814-0409f083a943)

Sometimes got this error
```
NiceGUI ready to go on http://localhost:8080, and http://192.168.1.113:8080
[hevc @ 0000027c0fd6e040] Could not find ref with POC 32
```

The video from rtsp doesn't refresh immediately，sometimes got this
```
NiceGUI ready to go on http://localhost:8080, and http://192.168.1.113:8080
[hevc @ 000001198f584500] The cu_qp_delta -45 is outside the valid range [-26, 25].
[hevc @ 000001198f5852c0] Could not find ref with POC 44
[rtsp @ 00000119e963c400] RTP: PT=60: bad cseq 17e7 expected=00d9
[hevc @ 000001198f3e7c40] Could not find ref with POC 8
```

Is it the problem of rtsp or nicegui?

**Comment by v479038280** [comment]
`https://blog.csdn.net/qq_43348528/article/details/109337221 `  this works

**Reply by v479038280** [reply]
> `https://blog.csdn.net/qq_43348528/article/details/109337221 ` this works

This also got some error
```
NiceGUI ready to go on http://localhost:8080, and http://192.168.1.113:8080
[ WARN:0@11572.493] global cap_ffmpeg_impl.hpp:453 _opencv_ffmpeg_interrupt_callback Stream timeout triggered after 30033.936000 ms
[ WARN:0@11602.600] global cap_ffmpeg_impl.hpp:453 _opencv_ffmpeg_interrupt_callback Stream timeout triggered after 30095.901000 ms
[ WARN:0@11632.775] global cap_ffmpeg_impl.hpp:453 _opencv_ffmpeg_interrupt_callback Stream timeout triggered after 30044.303000 ms
```

**Comment by rodja** [comment]
Are you sure this is a NiceGUI issue? Can you verify that the frames can be grabbed without problems? For example by writing them to disk instead of showing them in an interactive image?

**Reply by v479038280** [reply]
It's not the problem of nicegui ,it's because I'm not that familiar with cv2.Video_Capture('rtsp://')  and how to handle the frame within nicegui

**Reply by rodja** [reply]
If the problem occurs independently of NiceGUI, I suggest we close this issue here and you ask the opencv community to help you out.

**Comment by Deltaidiots** [comment]
I switched to av and its just smoother and better data handling would advise the same, In my case it was from an s3 bucket and i tried changing the flags from opencv to increase the timeout but that didnt solve anything for me
